prow/apis/prowjobs/v1/types.go

```diff
@@ -533,9 +533,9 @@ func (g *GCSConfiguration) ApplyDefault(def *GCSConfiguration) *GCSConfiguration
 	}
 	var merged GCSConfiguration
 	if g != nil {
-		merged = *g
+		merged = *g.DeepCopy()
```

Comment:
For the concrete issue at hand, doing the `DeepCopy` in the `ApplyDefaults` of `*DecorationConfig` is enough, but since this is public, there may be other callers that expect it to be threadsafe